Scheduler Updates / Search Prod Fixes

dhuangg opened the pull request:
Making updates to session CRUD and tooltip

dhuangg commented on src/components/FeatureViews/Scheduler/SchedulerUtils.js:
```diff
@@ -34,4 +36,87 @@ export const BootstrapInput = withStyles(theme => ({
+        const startTime = new Date(start).toTimeString();
+        const endTime = new Date(end).toTimeString();
+        // Converts 24hr to 12 hr time
+        function timeConverter(time) {
```
we should have a function for this in FormUtils.js and RegistrationCourse.js, check to see if they're being used. I don't think they are, so go ahead and delete them, but separate this function out to its own function.

dhuangg commented on src/components/FeatureViews/Scheduler/SchedulerUtils.js:
```diff
@@ -34,4 +36,87 @@ export const BootstrapInput = withStyles(theme => ({
+
+export const handleToolTip = (info) => {
+    function formatDate(start, end) {
+        const MonthConverter = {
```
Separate this out into its own object

dhuangg commented on src/components/FeatureViews/Scheduler/SchedulerUtils.js:
```diff
@@ -34,4 +36,87 @@ export const BootstrapInput = withStyles(theme => ({
+                            ${formatDate(info.event.start, info.event.end)}
+                        </span>
+                    </div>
+                    <div class='pin_icon icn'>
```
is "icn" actually a class?

dhuangg commented on src/components/FeatureViews/Scheduler/SchedulerUtils.js:
```diff
@@ -39,6 +39,18 @@ export const BootstrapInput = withStyles(theme => ({
 
 export const handleToolTip = (info) => {
+
+    function startAndEndDate(start, end) {
```
change this to an arrow function
```
const startAndEndDate = (start, end) => { ... }
```

NeilAllavarpu commented on src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js:
```diff
@@ -91,24 +98,48 @@ const RegistrationCourseEnrollments = ({courseID}) => {
 
+    const today = new Date();
+    const upcomingSession = currentMonthSessions
+        .filter((session) => ((Number(session.course) === Number(courseID)) &&
```
Can remove the type conversions and have them implicitly done with `==`

NeilAllavarpu commented on src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js:
```diff
@@ -91,24 +98,48 @@ const RegistrationCourseEnrollments = ({courseID}) => {
+            const distanceA = Math.abs(today - sessionA);
+            const distanceB = Math.abs(today - sessionB);
+            return distanceA - distanceB;
+        })[0];
```
Is this just trying to find the earliest date? You can simple sort by `sessionA - sessionB` if so.

NeilAllavarpu commented on src/components/FeatureViews/Scheduler/SessionFilters.js:
```diff
@@ -51,9 +51,7 @@ function SessionFilters({ onInstructorSelect, InstructorValue, InstructorOptions
-            <MenuItem
-                disabled={true}
-            > Select Filter</MenuItem>
+            <MenuItem selected={true}> Select Filter</MenuItem>
```
If a prop is equal to the boolean `true`, the `={true}` can be omitted (it defaults to the value `true`):

    <MenuItem selected>...

NeilAllavarpu commented on src/components/FeatureViews/Search/Search.js:
```diff
@@ -54,6 +54,9 @@ const Search = (props) => {
+        return () =>{
+            sessionStorage.setItem("SearchQuery",SearchQuery);
+        }
     },[]);
```
Should this be run only once or whenever the query changes?

dhuangg replied:
only whenever the query is changed. But the query isn't changed in Redux until the entire word has been typed out.

NeilAllavarpu replied:
In this case `SearchQuery` should probably be inside the dependencies array.

NeilAllavarpu commented on src/components/FeatureViews/Search/Search.js:
```diff
@@ -82,9 +85,21 @@ const Search = (props) => {
+                    toUpdate = true;
+                    return baseConfig
+                }
+            });
```
Can this be replaced with:

    toUpdate = /* if condition */
    setAccountRequestConfig(baseConfig);

?

NeilAllavarpu commented on src/components/FeatureViews/Search/Search.js:
```diff
@@ -112,9 +127,20 @@ const Search = (props) => {
+                if(JSON.stringify(courseRequestConfig) !== JSON.stringify(baseConfig)){
+                    toUpdate = true;
+                    return baseConfig
+                }
```
Same as above

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -33,120 +33,35 @@ const SearchResults = (props) => {
-            api.resetSearchParams();
+            // api.resetSearchParams();
         }
     }, []);
```
Delete the unused `useEffect`

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -156,31 +71,72 @@ const SearchResults = (props) => {
+
+    const setPageLimit = (resultsPageNumber) => {
+        return resultsPageNumber >= MAX_PAGE;
+    };
```
1. This is not `set`ting anything, so the function should be renamed appropriately.
2. You can use implicit returns from arrow functions here:

    const isWithinPageLimit = (resultsPageNumber) => (resultsPageNumber >= MAX_PAGE);

(parentheses around the condition are optional but added for clarity).

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -156,31 +71,72 @@ const SearchResults = (props) => {
+            return 0;
+        }
+
+    };
```
const startPage = (page) => searchState.primaryFilter === SEARCH_ALL && !isOdd(page) ? 4 : 0;

Also consider making page size a variable, if that might be changed in the future (the `4`)

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -156,31 +71,72 @@ const SearchResults = (props) => {
-            return Number(apiPage)*2+1;
+            return 8;
         }
     };
```
const endPage = (page) => startPage(page) + 4;

or you can rewrite it like the previous comment

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -156,31 +71,72 @@ const SearchResults = (props) => {
+            [resultType] : currentPage[resultType] + 1,
+        });
+
+        if(isOdd(page+1) || searchState.primaryFilter !== SEARCH_ALL){
```
isOdd(page + 1) === !isOdd(page)

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -156,31 +71,72 @@ const SearchResults = (props) => {
+            [resultType] : currentPage[resultType] - 1,
+        });
+
+        if(!isOdd(page-1) || searchState.primaryFilter !== SEARCH_ALL){
```
See above

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -223,40 +179,40 @@ const SearchResults = (props) => {
+                                    accountResultsNum > 0 && searchState.primaryFilter !== SEARCH_COURSES ?
+                                        accounts.slice(startPage(currentPage.account), endPage(currentPage.account))
+                                            .map((account) => {
+                                                // console.log(account)
```
Remove console.log

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -223,40 +179,40 @@ const SearchResults = (props) => {
-                                        :
-                                        ""}
+                                {
+                                    accountResultsNum > 0 && searchState.primaryFilter !== SEARCH_COURSES ?
```
Can replace ternary with an alternate value of `""` with an `&&`, in JSX (if the condition is false, nothing will be rendered)

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResults.js:
```diff
@@ -275,7 +231,7 @@ const SearchResults = (props) => {
                                 </Grid> : ""
                         }
-                        {account_num_results && course_num_results !== 0&&searchState.primaryFilter !== SEARCH_ACCOUNTS? <hr /> : ""}
+                        {accountResultsNum && courseResultsNum !== 0&&searchState.primaryFilter !== SEARCH_ACCOUNTS? <hr /> : ""}
```
`accountResultsNum` as a boolean condition checks against all falsey values, including 0, so the second condition is unnecessary.

NeilAllavarpu commented on src/components/FeatureViews/Search/cards/AccountsCards.js:
```diff
@@ -1,27 +1,23 @@
-    }
+        history.push(route);
+    };
     const stringToColor = (string) => {
```
Use the `stringToColor` in the utils file.

NeilAllavarpu commented on src/components/FeatureViews/Search/cards/AccountsCards.js:
```diff
@@ -55,18 +51,29 @@ function AccountsCards(props) {
+        </Grid>
+    }
 
+    const fullName = `${user.user.first_name} ${user.user.last_name}`;
```
`user.name` should be the appropriate property

NeilAllavarpu commented on src/components/FeatureViews/Search/cards/CoursesCards.js:
```diff
@@ -1,44 +1,60 @@
         e.preventDefault();
 
-        let courseID = props.course.course_id ? props.course.course.id : props.course.id;
+        let courseID = course.course_id ? course.course.id : course.id;
```
let courseID = course.course_id || course.id;

NeilAllavarpu commented on src/components/Form/Form.js:
```diff
@@ -898,18 +906,16 @@ class Form extends Component {
+                    this.props.currentParent.student_list.forEach((studentID) => {
+                        if (this.props.students[studentID]) {
+                            let { user_id, name, email } = this.props.students[studentID];
+                            studentList.push({
```
The `if` and `push` can be replaced with an appropriate `.filter()` chained to a `.map()`

dhuangg replied:
This can't because the if is checking to see if the student exists in redux. if it doesn't exist in redux, then don't do anything. with that said, i think this can be taken out because in a later commit in this pull I made it so that we're fetching all the parent's students.

NeilAllavarpu commented on src/components/Form/FormUtils.js:
```diff
@@ -132,12 +135,48 @@ export const gradeConverter = (grade) => {
+        time = to12HourTime + time.substr(2, 3) + ampm;
+        return time;
+
     }
```
I think we have a function like this somewhere else, not sure.

NeilAllavarpu commented on src/components/SessionPaymentStatusChip.js:
```diff
@@ -1,13 +1,22 @@
         </div>
     }
-    return <></>
+    return <div>hi</div>
```
Instead of having this you could have just made the component use an `if/else` lol.

dhuangg replied:
it. works. :|

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResultsLoader.js:
```diff
@@ -0,0 +1,119 @@
+                                                           key={account} />
+                                                   </Grid>
+                                               )
+                                            })
```
Use implicit returns.

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResultsLoader.js:
```diff
@@ -0,0 +1,119 @@
+                                                isLoading={true}
+                                                key={course} />
+                                        )}
+                                    )}
```
Use implicit returns.

dhuangg commented:
i'm just noticing, for some reason this PR is saying i added lines i never
added lol. i definitely didn't put the colorToString() function there and
some of the other lines LOL. will fix in the AM

On Sat, Mar 7, 2020 at 10:31 PM Neil Allavarpu <notifications@github.com>
wrote:

> *@NeilAllavarpu* requested changes on this pull request.
> ------------------------------
>
> In src/components/FeatureViews/Scheduler/EditSessionView.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r388651928>:
>
> >                      is_confirmed: is_confirmed,
>                      instructor: instructor.value,
>                      duration: duration,
> +                    title: title,
>
> When the key name matches the variable you are setting it to, it can be
> abbreviated:
>
> {
>     is_confirmed,
>     instructor: instructor.value,
>     duration,
>     title,
> }
>
> ------------------------------
>
> In
> src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389338537>:
>
> >          }
>          if (hooks.isFail(studentStatus)) {
>              return "Error loading enrollment details!";
>          }
>      }
>
> +    const today = new Date();
> +    const upcomingSession = currentMonthSessions
> +        .filter((session) => ((Number(session.course) === Number(courseID)) &&
>
> Can remove the type conversions and have them implicitly done with ==
> ------------------------------
>
> In
> src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389338614>:
>
> > +        .sort((sessionA, sessionB) => {
> +            const distanceA = Math.abs(today - sessionA);
> +            const distanceB = Math.abs(today - sessionB);
> +            return distanceA - distanceB;
> +        })[0];
>
> Is this just trying to find the earliest date? You can simple sort by sessionA
> - sessionB if so.
> ------------------------------
>
> In src/components/FeatureViews/Scheduler/SessionFilters.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389338799>:
>
> > @@ -51,9 +51,7 @@ function SessionFilters({ onInstructorSelect, InstructorValue, InstructorOptions
>              className={"session-filter"}
>              onClose={handleClick}
>          >
> -            <MenuItem
> -                disabled={true}
> -            > Select Filter</MenuItem>
> +            <MenuItem selected={true}> Select Filter</MenuItem>
>
> If a prop is equal to the boolean true, the ={true} can be omitted (it
> defaults to the value true):
>
> <MenuItem selected>...
>
> ------------------------------
>
> In src/components/FeatureViews/Search/Search.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389338862>:
>
> > @@ -54,6 +54,9 @@ const Search = (props) => {
>          if(SearchQuery !== prevSearchQuery){
>              api.setSearchQuery(prevSearchQuery);
>          }
> +        return () =>{
> +            sessionStorage.setItem("SearchQuery",SearchQuery);
> +        }
>      },[]);
>
> Should this be run only once or whenever the query changes?
> ------------------------------
>
> In src/components/FeatureViews/Search/Search.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389338950>:
>
> > +            setAccountRequestConfig(() => {
> +                if(JSON.stringify(accountRequestConfig) !== JSON.stringify(baseConfig)){
> +                    toUpdate = true;
> +                    return baseConfig
> +                }
> +            });
>
> Can this be replaced with:
>
> toUpdate = /* if condition */
> setAccountRequestConfig(baseConfig);
>
> ?
> ------------------------------
>
> In src/components/FeatureViews/Search/Search.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389338975>:
>
> > +            setCourseRequestConfig(()=>{
> +                if(JSON.stringify(courseRequestConfig) !== JSON.stringify(baseConfig)){
> +                    toUpdate = true;
> +                    return baseConfig
> +                }
>
> Same as above
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339019>:
>
> >      useEffect(() => {
>          return () => {
> -            api.resetSearchParams();
> +            // api.resetSearchParams();
>          }
>      }, []);
>
> Delete the unused useEffect
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339167>:
>
> > +    const setPageLimit = (resultsPageNumber) => {
> +        return resultsPageNumber >= MAX_PAGE;
> +    };
>
>
>    1.
>
>    This is not setting anything, so the function should be renamed
>    appropriately.
>    2.
>
>    You can use implicit returns from arrow functions here:
>
>    const isWithinPageLimit = (resultsPageNumber) => (resultsPageNumber >=
>    MAX_PAGE);
>
> (parentheses around the condition are optional but added for clarity).
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339252>:
>
> > +        if(searchState.primaryFilter === SEARCH_ALL){
> +            return isOdd(page) ? 0 : 4;
> +        } else {
> +            return 0;
> +        }
> +
> +    };
>
> const startPage = (page) => searchState.primaryFilter === SEARCH_ALL && !isOdd(page) ? 4 : 0;
>
> Also consider making page size a variable, if that might be changed in the
> future (the 4)
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339326>:
>
> > +    const endPage = (page) => {
> +        if(searchState.primaryFilter === SEARCH_ALL){
> +            return isOdd(page) ? 4 : 8;
>          } else {
> -            return Number(apiPage)*2+1;
> +            return 8;
>          }
>      };
>
> const endPage = (page) => startPage(page) + 4;
>
> or you can rewrite it like the previous comment
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339355>:
>
> >          }
>      };
>
> -    const setPageLimit = (resultsPageNumber) => {
> -        return resultsPageNumber >= MAX_PAGE;
> +    const handleMoreResults = (page, resultType) => (e) => {
> +        e.preventDefault();
> +        setCurrentPage({
> +            ...currentPage,
> +            [resultType] : currentPage[resultType] + 1,
> +        });
> +
> +        if(isOdd(page+1) || searchState.primaryFilter !== SEARCH_ALL){
>
> isOdd(page + 1) === !isOdd(page)
>
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339367>:
>
> >      };
> +
> +    const handleLessResults = (page, resultType) => (e) => {
> +        e.preventDefault();
> +        setCurrentPage({
> +            ...currentPage,
> +            [resultType] : currentPage[resultType] - 1,
> +        });
> +
> +        if(!isOdd(page-1) || searchState.primaryFilter !== SEARCH_ALL){
>
> See above
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339379>:
>
> > -                                    :
> -                                    account_num_results > 0 && searchState.primaryFilter !== SEARCH_COURSES?
> -                                        accounts.slice(start["account"], end["account"]).map((account) => (
> -                                            <Grid item
> -                                                  key={account.user_id}
> -                                                sm={3}>
> -                                                <AccountsCards user={account} key={account.user_id} />
> -                                            </Grid>
> -                                        ))
> -                                        :
> -                                        ""}
> +                                {
> +                                    accountResultsNum > 0 && searchState.primaryFilter !== SEARCH_COURSES ?
> +                                        accounts.slice(startPage(currentPage.account), endPage(currentPage.account))
> +                                            .map((account) => {
> +                                                // console.log(account)
>
> Remove console.log
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339421>:
>
> > -                                            <AccountsCards user={account} key={account.user_id} />
> -                                        </Grid>
> -                                    ))
> -                                    :
> -                                    account_num_results > 0 && searchState.primaryFilter !== SEARCH_COURSES?
> -                                        accounts.slice(start["account"], end["account"]).map((account) => (
> -                                            <Grid item
> -                                                  key={account.user_id}
> -                                                sm={3}>
> -                                                <AccountsCards user={account} key={account.user_id} />
> -                                            </Grid>
> -                                        ))
> -                                        :
> -                                        ""}
> +                                {
> +                                    accountResultsNum > 0 && searchState.primaryFilter !== SEARCH_COURSES ?
>
> Can replace ternary with an alternate value of "" with an &&, in JSX (if
> the condition is false, nothing will be rendered)
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResults.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339486>:
>
> > @@ -275,7 +231,7 @@ const SearchResults = (props) => {
>                                      </Grid>
>                                  </Grid> : ""
>                          }
> -                        {account_num_results && course_num_results !== 0&&searchState.primaryFilter !== SEARCH_ACCOUNTS? <hr /> : ""}
> +                        {accountResultsNum && courseResultsNum !== 0&&searchState.primaryFilter !== SEARCH_ACCOUNTS? <hr /> : ""}
>
> accountResultsNum as a boolean condition checks against all falsey
> values, including 0, so the second condition is unnecessary.
> ------------------------------
>
> In src/components/FeatureViews/Search/cards/AccountsCards.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339523>:
>
> >      const goToRoute = (route) => {
> -        props.history.push(route);
> -    }
> +        history.push(route);
> +    };
>      const stringToColor = (string) => {
>
> Use the stringToColor in the utils file.
> ------------------------------
>
> In src/components/FeatureViews/Search/cards/AccountsCards.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339545>:
>
> >
> +    const fullName = `${user.user.first_name} ${user.user.last_name}`;
>
> user.name should be the appropriate property
> ------------------------------
>
> In src/components/FeatureViews/Search/cards/CoursesCards.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339587>:
>
> >
>      const goToCoursePage = ()=> (e)=>{
>          e.preventDefault();
>
> -        let courseID = props.course.course_id ? props.course.course.id : props.course.id;
> +        let courseID = course.course_id ? course.course.id : course.id;
>
> let courseID = course.course_id || course.id;
>
> ------------------------------
>
> In src/components/Form/Form.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339656>:
>
> > +                        if (this.props.students[studentID]) {
> +                            let { user_id, name, email } = this.props.students[studentID];
> +                            studentList.push({
>
> The if and push can be replaced with an appropriate .filter() chained to
> a .map()
> ------------------------------
>
> In src/components/Form/FormUtils.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339747>:
>
> > +    function timeConverter(time) {
> +        const Hour = time.substr(0, 2);
> +        const to12HourTime = (Hour % 12) || 12;
> +        const ampm = Hour < 12 ? " am" : " pm";
> +        time = to12HourTime + time.substr(2, 3) + ampm;
> +        return time;
> +
>      }
>
> I think we have a function like this somewhere else, not sure.
> ------------------------------
>
> In src/components/SessionPaymentStatusChip.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389339885>:
>
> >              {status}
>          </div>
>      }
> -    return <></>
> +    return <div>hi</div>
>
> Instead of having this you could have just made the component use an
> if/else lol.
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResultsLoader.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389340042>:
>
> > +                                    [1,2,3,4].map((account) => {
> +                                                // console.log(account)
> +                                               return (
> +                                                   <Grid item
> +                                                         key={account}
> +                                                         sm={3}>
> +                                                       <AccountsCards
> +                                                           isLoading={true}
> +                                                           key={account} />
> +                                                   </Grid>
> +                                               )
> +                                            })
>
> Use implicit returns.
> ------------------------------
>
> In src/components/FeatureViews/Search/SearchResultsLoader.js
> <https://github.com/omou-org/front-end/pull/276#discussion_r389340069>:
>
> > +                                {[1,2,3,4].map((course) => {
> +                                        return (
> +                                            <CoursesCards
> +                                                isLoading={true}
> +                                                key={course} />
> +                                        )}
> +                                    )}
>
> Use implicit returns.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/omou-org/front-end/pull/276?email_source=notifications&email_token=ADC4BV47AXDY3255P72XOEDRGM3R7A5CNFSM4LAXMGT2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCYHATSI#pullrequestreview-370018761>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADC4BV2XDFIKJCSVTL35PJTRGM3R7ANCNFSM4LAXMGTQ>
> .
>


-- 
Daniel Huang
BA Economics | UCLA Class of 2018
danielhuang@ucla.edu | t: 925-858-3958

NeilAllavarpu commented on src/components/FeatureViews/Accounts/Accounts.js:
```diff
@@ -274,15 +274,11 @@ class Accounts extends Component {
-        ) {
-            return (<Loading />)
-        }
+            this.props.requestStatus.parent[GET][REQUEST_ALL] !== 200);
```
You can utilize `isLoading` in `hooks.js` (not an actual hook):

   const isLoading = hooks.isLoading(instructorStatus, studentStatus, parentStatus);

NeilAllavarpu commented on src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js:
```diff
@@ -91,24 +98,44 @@ const RegistrationCourseEnrollments = ({courseID}) => {
+            },
+            "id": "",
+        });
+    },[api]);
```
You can utilize the `useClassSessionsInPeriod/useTutoringSessionsInPeriod` hook in place of this, in hooks.js, or modify/create your own to fetch both at once (refer to code for how to do this).

NeilAllavarpu commented on src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js:
```diff
@@ -91,24 +98,44 @@ const RegistrationCourseEnrollments = ({courseID}) => {
 
+    const today = new Date();
+    const upcomingSession = currentMonthSessions
+        .filter((session) => ((Number(session.course) == Number(courseID)) &&
```
The conversions to `Number` are not necessary with `==`.

NeilAllavarpu commented on src/components/FeatureViews/Scheduler/SessionFilters.js:
```diff
@@ -1,11 +1,11 @@
 import blue from "@material-ui/core/es/colors/blue";
 import ReactSelect from "react-select";
-import { Tooltip, Typography, MenuItem } from "@material-ui/core";
+import {MenuItem, Tooltip} from "@material-ui/core";
```
Separately import these:

    import MenuItem from "@material-ui/core/MenuItem";
    import ToolTip from "@material-ui/core/Tooltip";

NeilAllavarpu commented on src/components/Form/Form.js:
```diff
@@ -1418,8 +1425,16 @@ class Form extends Component {
+        if(this.props.currentParent){
+            const studentsLoaded = this.props.currentParent.student_list.every((student)=>{
+                return this.props.requestStatus.student[GET][student] === 200;
+            });
```
(Optional) Use implicit arrow function returns.

NeilAllavarpu commented on src/utils.js:
```diff
@@ -95,6 +98,35 @@ export const courseToRegister = (enrollment, course, student) => ({
+
+const isoStringToUTCDate = (dateString) => {
+    let newDate = new Date("2020-01-01");
+    console.log(Number(dateString.substring(5,7)));
```
Remove `console.log`

NeilAllavarpu commented on src/utils.js:
```diff
@@ -95,6 +98,35 @@ export const courseToRegister = (enrollment, course, student) => ({
+    } else {
+        startDate = start.toISOString().substring(0,start.toISOString().indexOf("T"));
+        endDate = end.toISOString().substring(0,end.toISOString().indexOf("T"));
+        console.log(isoStringToUTCDate(startDate));
```
Remove `console.log`

NeilAllavarpu commented on src/utils.js:
```diff
@@ -95,6 +98,35 @@ export const courseToRegister = (enrollment, course, student) => ({
+    newDate.setFullYear(Number(dateString.substring(0,4)));
+    newDate.setMonth(Number(dateString.substring(5,7)));
+    newDate.setDate(Number(dateString.substring(8,10)));
+
```
None of this code does anything, as it does not perform side effects nor is it used in the return value.

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResultsLoader.js:
```diff
@@ -0,0 +1,114 @@
+                                                         key={account}
+                                                         sm={3}>
+                                                       <AccountsCards
+                                                           isLoading={true}
```
Replace with simply `isLoading`

NeilAllavarpu commented on src/components/FeatureViews/Search/SearchResultsLoader.js:
```diff
@@ -0,0 +1,114 @@
+                            <Grid container spacing={8} direction={"row"}>
+                                {[1,2,3,4].map((course) => (
+                                            <CoursesCards
+                                                isLoading={true}
```
Replace with `isLoading`

NeilAllavarpu commented on .env.local:
```diff
@@ -1,3 +1,3 @@
-#REACT_APP_DOMAIN = http://api.omoulearning.net
\ No newline at end of file
+#REACT_APP_DOMAIN = http://localhost:8000
+REACT_APP_DOMAIN = http://api.omoulearning.net
```
Undo this.

NeilAllavarpu commented on src/components/FeatureViews/Accounts/Accounts.js:
```diff
@@ -17,19 +16,21 @@ import TableCell from "@material-ui/core/TableCell";
 import {REQUEST_ALL} from "../../../actions/apiActions";
 import Loading from "../../Loading";
-import { Tooltip } from "@material-ui/core";
+import {isLoading} from "../../../actions/hooks"
```
Can be imported `... from "actions/hooks"`

NeilAllavarpu commented on src/components/FeatureViews/Accounts/Accounts.js:
```diff
@@ -17,19 +16,21 @@ import TableCell from "@material-ui/core/TableCell";
 import Loading from "../../Loading";
-import { Tooltip } from "@material-ui/core";
+import {isLoading} from "../../../actions/hooks"
+import {Tooltip} from "@material-ui/core";
```
Import as

    import Tooltip from "@material-ui/core/Tooltip";

NeilAllavarpu commented on src/components/FeatureViews/Accounts/Accounts.js:
```diff
@@ -200,26 +199,19 @@ class Accounts extends Component {
-                            <TableCell
-                                style={{ width: 150 }}>
+                            <TableCell>
                                 {row.role.charAt(0).toUpperCase() + row.role.slice(1)}
```
Use the capitalization function if it is available (may need to merge in development).

NeilAllavarpu commented on src/components/FeatureViews/Accounts/Accounts.js:
```diff
@@ -228,13 +220,12 @@ class Accounts extends Component {
-                                        <Button
-                                            className="editButton"
+                                        <IconButton
+                                            // className="editButton"
```
Remove comment

NeilAllavarpu commented on src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js:
```diff
@@ -23,6 +23,7 @@ import MenuItem from "@material-ui/core/MenuItem";
 import IconButton from "@material-ui/core/IconButton";
 import {bindActionCreators} from "redux";
 import * as registrationActions from "../../../actions/registrationActions";
+import * as calendarActions from "../../../actions/calendarActions";
```
Can be imported as `... from "actions/____Actions";`

NeilAllavarpu commented on src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js:
```diff
@@ -31,11 +32,14 @@ import DialogActions from "@material-ui/core/DialogActions";
 import {NoListAlert} from "../../NoListAlert";
+import {sessionArray} from "../Scheduler/SchedulerUtils";
+import {SessionPaymentStatusChip} from "../../SessionPaymentStatusChip";
+import {useSessionsInPeriod} from "../../../actions/hooks";
```
Can be imported as `... from "actions/hooks";`

NeilAllavarpu commented on src/components/FeatureViews/Registration/SelectParentDialog.js:
```diff
@@ -18,28 +18,29 @@ class SelectParentDialog extends React.Component {
             let parentID = Number(this.state.inputParent.value.substring(idStartIndex));
-            let selectedParent = this.props.search.accounts.find((account)=>{ return parentID === account.user.id});
-            if(selectedParent){
+            let selectedParent = this.state.parentList.find((account) => { return parentID == account.user.id });
```
Can use implicit arrow function returns

NeilAllavarpu commented on src/components/FeatureViews/Scheduler/SessionFilters.js:
```diff
@@ -1,11 +1,12 @@
 import ReactSelect from "react-select";
-import { Tooltip, Typography, MenuItem } from "@material-ui/core";
+import MenuItem from "@material-ui/core/es/MenuItem";
+import Tooltip from "@material-ui/core/es/Tooltip";
```
It is recommended to [avoid the `/es/` imports](https://material-ui.com/guides/minimizing-bundle-size/#ecmascript)

NeilAllavarpu commented on src/reducers/searchResultsReducer.js:
```diff
@@ -124,7 +126,7 @@ const handleSearchFilterChange = (state, { searchType, filter, value }) => {
         newState.params[searchType].profile = "student";
     }
-    newState.searchQueryStatus = "";
+    // newState.searchQueryStatus = "";
```
Remove comment

NeilAllavarpu commented on src/reducers/searchResultsReducer.js:
```diff
@@ -74,9 +76,9 @@ export default function course(state = initialState.SearchResults, { payload, ty
     let { response } = payload;
     let { data } = response;
-
+    console.log("account search updating!", data);
```
Remove `console.log`

NeilAllavarpu commented on src/components/FeatureViews/Search/Search.js:
```diff
@@ -82,9 +85,21 @@ const Search = (props) => {
+                    return baseConfig
+                }
+            });
+            // api.updateSearchStatus(NOT_SEARCHING);
```
Remove comment

NeilAllavarpu commented on src/components/FeatureViews/Search/Search.js:
```diff
@@ -206,25 +224,30 @@ const Search = (props) => {
         }
+        api.updateSearchStatus(IS_SEARCHING);
+        // api.updateSearchParam("account", "accountPage", 1);
+        // api.updateSearchParam("course", "coursePage", 1);
```
Remove comment.

NeilAllavarpu commented on src/components/Form/Form.js:
```diff
@@ -799,7 +808,10 @@ class Form extends Component {
-                                        value={option}>
+                                        value={option}
+                                        selected={option === "1 Hour" && fieldTitle === "Duration" ? true :
+                                            fieldTitle !== "Duration"}
```
Can be replaced with `fieldDuration !== "Duration" || option === "1 hour"` (Is this why we can only select 1 hour options?)

dhuangg replied:
lol no, I fixed that but I was trying to do something slick here to show the selected option as 1 hour instead of 0.5 hour

NeilAllavarpu commented on src/components/Form/FormUtils.js:
```diff
@@ -1,10 +1,12 @@
 import {instance} from "actions/apiActions";
 import {isFail} from "actions/hooks";
+import {DayConverter} from "utils";
+import {capitalizeString} from "../../utils";
```
Merge these imports, they are from the same file.

dhuangg commented on src/actions/registrationActions.js:
```diff
@@ -173,22 +173,17 @@ export const submitForm = (state, id) => {
+            const course = formatCourse(state, "class");
 
             for (const key in course) {
                 if (course.hasOwnProperty(key) && !course[key]) {
```
I think that's fine. either way they'll be defaulted to something null/undefined
